plotting/comparison.py

```diff
@@ -239,89 +262,58 @@
                 label=s + "-b",
                 yerr=True,
                 color="r",
-                marker=markers[index + 1],
+                marker=markers[mindex + 1],
                 histtype="errorbar",
                 ax=ax,
             )
-        index += 1
+            mindex += 1
         ax.legend(
             ncol=3,
             loc=1,
         )
-        index = 0
+        # comparison splitted by flavor
         for c in args.compared.split(","):
-
-            rax.errorbar(
-                x=collated[c][discr][laxis].axes[0].centers,
-                y=(
-                    collated[c][discr][laxis].values()
-                    + collated[c][discr][puaxis].values()
-                )
-                / (
-                    collated[args.ref][discr][laxis].values()
-                    + collated[args.ref][discr][puaxis].values()
-                ),
-                yerr=ratio_uncertainty(
-                    (
-                        collated[c][discr][laxis].values()
-                        + collated[c][discr][puaxis].values()
-                    ),
-                    collated[args.ref][discr][laxis].values()
-                    + collated[args.ref][discr][puaxis].values(),
-                ),
-                color="b",
-                linestyle="none",
-                marker=markers[index + 1],
+            rax = plotratio(
+                collated[c][discr][laxis] + collated[c][discr][puaxis],
+                collated[args.ref][discr][laxis] + collated[args.ref][discr][puaxis],
+                ax=rax,
+                denom_fill_opts=None,
+                error_opts={"color": "b", "marker": markers[index + 1]},
             )
-            rax2.errorbar(
-                x=collated[c][discr][caxis].axes[0].centers,
-                y=collated[c][discr][caxis].values()
-                / collated[args.ref][discr][caxis].values(),
-                yerr=ratio_uncertainty(
-                    collated[c][discr][caxis].values(),
-                    collated[args.ref][discr][caxis].values(),
-                ),
-                color="g",
-                linestyle="none",
-                marker=markers[index + 1],
+            rax2 = plotratio(
+                collated[c][discr][caxis],
+                collated[args.ref][discr][caxis],
+                ax=rax2,
+                denom_fill_opts=None,
+                error_opts={"color": "g", "marker": markers[index + 1]},
```

Comment:
Add `mindex = 0` before start of the loop and change to `markers[mindex + 1]` everywhere